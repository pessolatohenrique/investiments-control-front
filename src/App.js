import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { isPageWithoutLogin } from "./utils/pages";
import useToken from "./hooks/useToken";
import AppBarWrapper from "./components/AppBarWrapper";
import NotFound from "./components/NotFound";
import LoginForm from "./views/register/LoginForm";
import RegisterContainer from "./views/register/RegisterContainer";
import DashboardContainer from "./views/dashboard/DashboardContainer";
import RecipeContainer from "./views/recipe/RecipeContainer";
import InvestimentContainer from "./views/investiments/InvestimentContainer";
import EarningForm from "./views/earnings/EarningForm";
import ExpenseForm from "./views/expenses/ExpenseForm";
import InvestimentForm from "./views/investiments/InvestimentForm";

function App(props) {
  const { token, setToken, setRefreshToken } = useToken();

  if (!token && !isPageWithoutLogin()) {
    return <LoginForm setToken={setToken} setRefreshToken={setRefreshToken} />;
  }

  return (
    <Router>
      <AppBarWrapper />

      <Switch>
        <Route exact path="/">
          <DashboardContainer />
        </Route>

        <Route exact path="/investimentos">
          <InvestimentContainer />
        </Route>

        <Route exact path="/investimento/novo">
          <InvestimentForm />
        </Route>

        <Route exact path="/investimento/:id">
          <InvestimentForm />
        </Route>

        <Route exact path="/receita/associar">
          <RecipeContainer />
        </Route>

        <Route exact path="/usuario/novo">
          <RegisterContainer />
        </Route>

        <Route exact path="/ganho/novo">
          <EarningForm />
        </Route>

        <Route exact path="/ganho/:id">
          <EarningForm />
        </Route>

        <Route exact path="/gasto/novo">
          <ExpenseForm />
        </Route>

        <Route exact path="/gasto/:id">
          <ExpenseForm />
        </Route>

        <Route path="*">
          <NotFound />
        </Route>
      </Switch>
    </Router>
  );
}

export default App;
