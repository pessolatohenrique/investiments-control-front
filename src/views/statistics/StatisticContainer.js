import React, { useState, useEffect } from "react";
import axios from "axios";
import {
  Grid,
  Card,
  CardActions,
  CardContent,
  Button,
  Typography,
  Alert,
  Snackbar,
  Box,
} from "@mui/material";
import { THEME_COLOR } from "../../constants/default_settings";
import ViewListToggle from "../../components/ViewListToggle";
import { BarChartComparative } from "../../components/ChartWrapper";
import useView from "../../hooks/useView";
import { GoalList } from "../goals/GoalList";
import { StatisticTable } from "../statistics/StatisticTable";

import { SNACKBAR_DIRECTION } from "../../constants/default_settings";
import useToast from "../../hooks/useToast";

function useViewWrapper(initial = "table") {
  const [isTable, isList, switchFormat] = useView(initial);
  return [isTable, isList, switchFormat];
}

function StatisticContainer() {
  // custom hooks
  const { open, error, setError, showToast, hideToast } = useToast();

  const [
    isTableStatisticType,
    isListStatisticType,
    switchStatisticType,
  ] = useViewWrapper("list");
  const [
    isTableStatisticYear,
    isListStatisticYear,
    switchStatisticYear,
  ] = useViewWrapper("list");
  const [
    isTableStatisticPlatform,
    isListStatisticPlatform,
    switchStatisticPlatform,
  ] = useViewWrapper("list");
  const [
    isTableStatisticCategory,
    isListStatisticCategory,
    switchStatisticCategory,
  ] = useViewWrapper("list");

  // local states
  const [statisticsByType, setStatisticsByType] = useState([]);
  const [statisticsByYear, setStatisticsByYear] = useState([]);
  const [statisticsByPlatform, setStatisticsByPlatform] = useState([]);
  const [statisticsByCategory, setStatisticsByCategory] = useState([]);

  async function getStatisticsByType() {
    try {
      const response = await axios.get(`/statistic/group?by=type`);
      setStatisticsByType(response?.data?.result);
    } catch (error) {
      showToast();
      setError(error?.response?.data?.message || null);
    }
  }

  async function getStatisticsByYear() {
    try {
      const response = await axios.get(`/statistic/group?by=final_date`);
      setStatisticsByYear(response?.data?.result);
    } catch (error) {
      showToast();
      setError(error?.response?.data?.message || null);
    }
  }

  async function getStatisticsByPlatform() {
    try {
      const response = await axios.get(`/statistic/group?by=platform`);
      setStatisticsByPlatform(response?.data?.result);
    } catch (error) {
      showToast();
      setError(error?.response?.data?.message || null);
    }
  }

  async function getStatisticsByCategory() {
    try {
      const response = await axios.get(`/statistic/group?by=category`);
      setStatisticsByCategory(response?.data?.result);
    } catch (error) {
      showToast();
      setError(error?.response?.data?.message || null);
    }
  }

  useEffect(() => {
    getStatisticsByType();
    getStatisticsByYear();
    getStatisticsByPlatform();
    getStatisticsByCategory();
  }, []);

  return (
    <Box
      minHeight="100vh"
      sx={{
        backgroundColor: `#E5E5E5`,
        overflow: "hidden",
      }}
    >
      <Snackbar
        anchorOrigin={SNACKBAR_DIRECTION}
        open={open}
        autoHideDuration={6000}
        onClose={hideToast}
      >
        <Alert onClose={hideToast} severity="error" sx={{ width: "100%" }}>
          {error}
        </Alert>
      </Snackbar>

      <Grid
        container
        columnSpacing={3}
        marginLeft={2}
        marginRight={2}
        marginTop={5}
        marginBottom={3}
        sx={{ width: "96%" }}
      >
        <Grid item lg={12} md={12} sm={12} xs={11}>
          <Card>
            <CardContent>
              <Typography
                gutterBottom
                variant="h5"
                component="h1"
                color={THEME_COLOR}
              >
                Por tipo de investimento
              </Typography>

              <ViewListToggle
                isTable={isTableStatisticType}
                isList={isListStatisticType}
                switchFormat={switchStatisticType}
              />

              {isTableStatisticType() && (
                <StatisticTable result={statisticsByType} />
              )}

              {isListStatisticType() && !process.env.JEST_WORKER_ID && (
                <BarChartComparative
                  data={statisticsByType}
                  labelProperty="dream_name"
                  mainConfig={{
                    legend: "Esperado (R$)",
                    backgroundColor: "rgba(53, 162, 235, 0.5)",
                    valueProperty: "sum_expected_net_value",
                  }}
                  secondConfig={{
                    legend: "Investido (R$)",
                    backgroundColor: "rgba(82, 178, 191, 0.5)",
                    valueProperty: "sum_invested_amount",
                  }}
                />
              )}
            </CardContent>
          </Card>
        </Grid>
      </Grid>

      <Grid
        container
        columnSpacing={3}
        marginLeft={2}
        marginRight={2}
        marginTop={5}
        marginBottom={3}
        sx={{ width: "96%" }}
      >
        <Grid item lg={12} md={12} sm={12} xs={11}>
          <Card>
            <CardContent>
              <Typography
                gutterBottom
                variant="h5"
                component="h1"
                color={THEME_COLOR}
              >
                Por ano de resgate
              </Typography>

              <ViewListToggle
                isTable={isTableStatisticYear}
                isList={isListStatisticYear}
                switchFormat={switchStatisticYear}
              />

              {isTableStatisticYear() && (
                <StatisticTable result={statisticsByYear} />
              )}

              {isListStatisticYear() && !process.env.JEST_WORKER_ID && (
                <BarChartComparative
                  data={statisticsByYear}
                  labelProperty="dream_name"
                  mainConfig={{
                    legend: "Esperado (R$)",
                    backgroundColor: "rgba(53, 162, 235, 0.5)",
                    valueProperty: "sum_expected_net_value",
                  }}
                  secondConfig={{
                    legend: "Investido (R$)",
                    backgroundColor: "rgba(82, 178, 191, 0.5)",
                    valueProperty: "sum_invested_amount",
                  }}
                />
              )}
            </CardContent>
          </Card>
        </Grid>
      </Grid>

      <Grid
        container
        columnSpacing={3}
        marginLeft={2}
        marginRight={2}
        marginTop={5}
        marginBottom={3}
        sx={{ width: "96%" }}
      >
        <Grid item lg={12} md={12} sm={12} xs={11}>
          <Card>
            <CardContent>
              <Typography
                gutterBottom
                variant="h5"
                component="h1"
                color={THEME_COLOR}
              >
                Por plataforma
              </Typography>

              <ViewListToggle
                isTable={isTableStatisticPlatform}
                isList={isListStatisticPlatform}
                switchFormat={switchStatisticPlatform}
              />

              {isTableStatisticPlatform() && (
                <StatisticTable result={statisticsByPlatform} />
              )}

              {isListStatisticYear() && !process.env.JEST_WORKER_ID && (
                <BarChartComparative
                  data={statisticsByPlatform}
                  labelProperty="dream_name"
                  mainConfig={{
                    legend: "Esperado (R$)",
                    backgroundColor: "rgba(53, 162, 235, 0.5)",
                    valueProperty: "sum_expected_net_value",
                  }}
                  secondConfig={{
                    legend: "Investido (R$)",
                    backgroundColor: "rgba(82, 178, 191, 0.5)",
                    valueProperty: "sum_invested_amount",
                  }}
                />
              )}
            </CardContent>
          </Card>
        </Grid>
      </Grid>

      <Grid
        container
        columnSpacing={3}
        marginLeft={2}
        marginRight={2}
        marginTop={5}
        marginBottom={3}
        sx={{ width: "96%" }}
      >
        <Grid item lg={12} md={12} sm={12} xs={11}>
          <Card>
            <CardContent>
              <Typography
                gutterBottom
                variant="h5"
                component="h1"
                color={THEME_COLOR}
              >
                Por categoria
              </Typography>

              <ViewListToggle
                isTable={isTableStatisticCategory}
                isList={isListStatisticCategory}
                switchFormat={switchStatisticCategory}
              />

              {isTableStatisticCategory() && (
                <StatisticTable result={statisticsByCategory} />
              )}

              {isListStatisticYear() && !process.env.JEST_WORKER_ID && (
                <BarChartComparative
                  data={statisticsByCategory}
                  labelProperty="dream_name"
                  mainConfig={{
                    legend: "Esperado (R$)",
                    backgroundColor: "rgba(53, 162, 235, 0.5)",
                    valueProperty: "sum_expected_net_value",
                  }}
                  secondConfig={{
                    legend: "Investido (R$)",
                    backgroundColor: "rgba(82, 178, 191, 0.5)",
                    valueProperty: "sum_invested_amount",
                  }}
                />
              )}
            </CardContent>
          </Card>
        </Grid>
      </Grid>
    </Box>
  );
}

export default StatisticContainer;
