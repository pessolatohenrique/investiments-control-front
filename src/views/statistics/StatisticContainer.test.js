import axios from "axios";
import {
  render,
  screen,
  fireEvent,
  waitForElement,
} from "@testing-library/react";
import App from "../../App";
import StatisticContainer from "./StatisticContainer";

jest.mock("axios");

function mockCallValid() {
  axios.get.mockResolvedValueOnce({
    data: [
      {
        dream_name: "rico",
        sum_expected_net_value: 1501.9,
        sum_invested_amount: 1091,
        expected_profit: 410.9,
      },
      {
        dream_name: "clear",
        sum_expected_net_value: 2560,
        sum_invested_amount: 2000,
        expected_profit: 560,
      },
    ],
  });
}

describe("Statistic Container", () => {
  beforeAll(() => {
    render(<App />);
    const submitBtn = screen.getByTestId("submit-button");
    const username = screen.getByTestId("username");
    const password = screen.getByTestId("password");

    fireEvent.input(username, {
      target: { value: process.env.REACT_APP_API_DEFAULT_USER },
    });
    fireEvent.input(password, {
      target: {
        value: process.env.REACT_APP_API_DEFAULT_EMAIL,
      },
    });
    fireEvent.click(submitBtn);
  });

  it("should generate snapshot", () => {
    const { container } = render(<StatisticContainer />);
    expect(container).toMatchSnapshot();
  });

  it("should render investiments after request", async () => {
    mockCallValid();
    render(<StatisticContainer />);

    await waitForElement(() => screen.getAllByText("Por plataforma"));
  });
});
