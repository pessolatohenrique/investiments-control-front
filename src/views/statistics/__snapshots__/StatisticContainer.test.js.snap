// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistic Container should generate snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-m4l5cg"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-13puk6r-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-25posd-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
            >
              Por tipo de investimento
            </h1>
            <div
              class="MuiGrid-root MuiGrid-container css-1mx4t5n-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                data-testid="view-table"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ListIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                data-testid="view-card"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ViewModuleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 11h5V5H4v6zm0 7h5v-6H4v6zm6 0h5v-6h-5v6zm6 0h5v-6h-5v6zm-6-7h5V5h-5v6zm6-6v6h5V5h-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-13puk6r-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-25posd-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
            >
              Por ano de resgate
            </h1>
            <div
              class="MuiGrid-root MuiGrid-container css-1mx4t5n-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                data-testid="view-table"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ListIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                data-testid="view-card"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ViewModuleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 11h5V5H4v6zm0 7h5v-6H4v6zm6 0h5v-6h-5v6zm6 0h5v-6h-5v6zm-6-7h5V5h-5v6zm6-6v6h5V5h-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-13puk6r-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-25posd-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
            >
              Por plataforma
            </h1>
            <div
              class="MuiGrid-root MuiGrid-container css-1mx4t5n-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                data-testid="view-table"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ListIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                data-testid="view-card"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ViewModuleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 11h5V5H4v6zm0 7h5v-6H4v6zm6 0h5v-6h-5v6zm6 0h5v-6h-5v6zm-6-7h5V5h-5v6zm6-6v6h5V5h-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-13puk6r-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-25posd-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
            >
              Por categoria
            </h1>
            <div
              class="MuiGrid-root MuiGrid-container css-1mx4t5n-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                data-testid="view-table"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ListIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                data-testid="view-card"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ViewModuleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 11h5V5H4v6zm0 7h5v-6H4v6zm6 0h5v-6h-5v6zm6 0h5v-6h-5v6zm-6-7h5V5h-5v6zm6-6v6h5V5h-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-13puk6r-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-25posd-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
            >
              Por indexador
            </h1>
            <div
              class="MuiGrid-root MuiGrid-container css-1mx4t5n-MuiGrid-root"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                data-testid="view-table"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ListIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                data-testid="view-card"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ViewModuleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 11h5V5H4v6zm0 7h5v-6H4v6zm6 0h5v-6h-5v6zm6 0h5v-6h-5v6zm-6-7h5V5h-5v6zm6-6v6h5V5h-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
